src/backend/cuda/platform.cpp

```diff
@@ -496,16 +496,26 @@ DeviceManager::DeviceManager()
     CUDA_CHECK(cudaGetDeviceCount(&nDevices));
     if (nDevices == 0)
         throw runtime_error("No CUDA-Capable devices found");
-
     cuDevices.reserve(nDevices);
 
+    int cudaRtVer = 0;
+    CUDA_CHECK(cudaRuntimeGetVersion(&cudaRtVer));
+    int cudaMajorVer = cudaRtVer / 1000;
+
     for(int i = 0; i < nDevices; i++) {
         cudaDevice_t dev;
         cudaGetDeviceProperties(&dev.prop, i);
-        dev.flops = dev.prop.multiProcessorCount * compute2cores(dev.prop.major, dev.prop.minor) * dev.prop.clockRate;
-        dev.nativeId = i;
-        cuDevices.push_back(dev);
+        if (cudaMajorVer>8 && dev.prop.major<=2) {
```

Comment:
It would be cleaner if this was implemented as a table of CUDA versions and minimum supported compute device. In the future CUDA will most likely deprecate additional hardware and that will make this implementation more complicated.

Reply:
what kind of a table are you talking about ? Will a future deprecation be so complicated ? It would probably add another check `cudaMajorVer > 10 && dev.prop.major<=3` or something like that to the top of the if-else block at the top.

Reply:
You can just a map or a vector that stores the minimum supported compute for each version of cuda. something like this:

vector<int> versions = {0, 0, 0, 0, 0, 0, 0, 00100, 00102};

or something similar.

Reply:
That is an overkill for just couple of checks. Vector or not, we can't eliminate the if condition here, it will only change the form into `dev.propr.major<=versions[cudaMajorVer-1]` which doesn't add any benefits because this versions vector isn't static and have to be updated every time CUDA toolkit drops a specific compute.

Added to that, current form is easier to understand and this code is not in a performance bottleneck zone.